src/components/Receiving/ItemDetails.js

```diff
@@ -75,11 +88,78 @@ class ItemDetails extends Component {
     });
   }
 
-  onClickNext = () => (
-    this.setState(({ currentLine }) => ({
-      currentLine: currentLine + 1,
-    }))
-  );
+  finalToggleAll = () => (
+    this.setState(state => {
+      const isFinalAllChecked = !state.isFinalAllChecked;
+      const finalCheckedItemList = [...state.finalCheckedItemList];
+
+      if (state.isFinalAllChecked) {
+        const toggledItems = finalCheckedItemList.map(item => {
+          item.isSelected = false;
+
+          return item;
+        });
+
+        return {
+          isFinalAllChecked,
+          finalCheckedItemList: toggledItems,
+        };
+      } else {
+        const toggledItems = finalCheckedItemList.map(item => {
+          item.isSelected = true;
+
+          return item;
+        });
+
+        return {
+          isFinalAllChecked,
+          finalCheckedItemList: toggledItems,
+        };
+      }
+    })
+  )
+
+  finalToggleItem = (item) => {
```

Comment:
what is the conceptual difference between toggling pieces on the review or piece details screen? I thought it's supposed to use the same logic.